### PR TITLE
Pass womtool endpoint through CromIAM

### PR DESCRIPTION
**Pull request by aednichols**
For use by Terra/Firecloud, in concert with https://github.com/broadinstitute/firecloud-orchestration/pull/710

Closes https://github.com/broadinstitute/cromwell/issues/4728

### CODE REVIEW

File: CromIAM/src/test/scala/cromiam/webservice/MockClients.scala [hunk 1]
```diff
@@ -51,11 +51,15 @@ class MockCromwellClient()(implicit system: ActorSystem,
 
   override def forwardToCromwell(httpRequest: HttpRequest): FailureResponseOrT[HttpResponse] = {
     val versionRoutePath = s"/engine/$version/version"
+    val womtoolRoutePath = s"/api/womtool/$version/describe"
 
     httpRequest.uri.path.toString match {
       //version endpoint doesn't require authentication
       case `versionRoutePath` =>
         FailureResponseOrT.pure(HttpResponse(status = OK, entity = "Response from Cromwell"))
+      // womtool endpoint requires authn which it gets for free from the proxy, does not care about authz
```
Comment (aednichols) [inline CromIAM/src/test/scala/cromiam/webservice/MockClients.scala]: Regardless of how many thumbs I have I am not going to merge this until I have personally received confirmation of this from @adrazhi

File: CromIAM/src/main/scala/cromiam/webservice/WomtoolRouteSupport.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+trait WomtoolRouteSupport extends RequestSupport {
+  // When this trait is mixed into `CromIamApiService` the value of `cromwellClient` is the reader (non-abort) address
+  val cromwellClient: CromwellClient
+
+  val womtoolRoutes =
+    path("api" / "womtool" / Segment / "describe") { _ =>
+      post {
+        extractStrictRequest { req =>
+          complete {
+            // This endpoint requires authn which it gets for free from the proxy, does not care about authz
```
Comment (aednichols) [inline CromIAM/src/main/scala/cromiam/webservice/WomtoolRouteSupport.scala]: Regardless of how many thumbs I have I am not going to merge this until I have personally received confirmation of this from @adrazhi

File: engine/src/main/resources/swagger/cromwell.yaml [hunk 3]
```diff
@@ -643,7 +642,7 @@ paths:
           schema:
             $ref: '#/definitions/WorkflowDescription'
       tags:
-        - "Womtool (beta)"
+        - "Womtool"
```
Comment (kshakir) [inline engine/src/main/resources/swagger/cromwell.yaml]: 🙌

File: CromIAM/src/main/resources/swagger/cromiam.yaml [hunk 4]
```diff
@@ -934,6 +994,252 @@ definitions:
...
+        type: boolean
+        description: Whether the workflow and inputs as submitted passed validation
+      errors:
+        type: array
+        items:
+          type: string
+        example:
+          - "The 'errors' field will be filled if 'valid' is false"
+          - "We might also provide warnings to a 'valid' workflow here"
+          - "Otherwise, 'errors' will be the empty array"
```
Comment (mcovarr) [inline CromIAM/src/main/resources/swagger/cromiam.yaml]: no programmatic way to discern warnings from errors?

File: CromIAM/src/main/resources/swagger/cromiam.yaml [hunk 5]
```diff
@@ -934,6 +994,252 @@ definitions:
...
+        description: The set of validation failure messages
```
Comment (mcovarr) [inline CromIAM/src/main/resources/swagger/cromiam.yaml]: and warnings

File: CromIAM/src/main/scala/cromiam/webservice/WomtoolRouteSupport.scala (hunk 2 above)
Comment (aednichols) [reply CromIAM/src/main/scala/cromiam/webservice/WomtoolRouteSupport.scala]: Proxy writeup: https://medium.com/@lukwam/open-sourcing-openidc-proxy-67cc3870f168

Comment (aednichols) [reply CromIAM/src/main/scala/cromiam/webservice/WomtoolRouteSupport.scala]: Confirmed!

**Pull request merged**